Prelude/Tuple.swift

```diff
@@ -1,5 +1,7 @@
 public func == <A: Equatable, B: Equatable> (lhs: [(A, B)], rhs: [(A, B)]) -> Bool {
-  for (idx, _) in lhs.enumerated() {
+  guard lhs.count == rhs.count else { return false }
+
+  for idx in lhs.indices {
```

Comment:
new swiftlint rule that catches if you use `enumerated()` when `indices` will do just fine.